src/bindings/tsd.ts

```diff
@@ -185,6 +185,11 @@ export class TSDBuilder extends ExportsWalker {
     if (!this.esm) {
       sb.push("declare namespace __AdaptedExports {\n");
       ++this.indentLevel;
+    } else {
+      if (this.program.options.exportMemory) {
+        sb.push("/** bindings/esm/memory */\n");
```

Comment:
Memory doesn't really have an internal name, and `bindings/esm` is not the name in general (in the existing tests it's because the entry file is `bindings/esm.ts`). Perhaps just...?
```suggestion
      sb.push("/** Exported memory */\n");
```